datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -796,18 +801,6 @@ impl<'a, S: SimplifyInfo> TreeNodeRewriter for Simplifier<'a, S> {
                 op: Divide,
                 right,
             }) if is_null(&right) => *right,
-            // A / 0 -> Divide by zero error if A is not null and not floating
-            // (float / 0 -> inf | -inf | NAN)
-            Expr::BinaryExpr(BinaryExpr {
```

Comment:
Remove this because `A / 0` can occur in short-circuit expression. And we can throw such an error in runtime.